src/senaite/sync/syncstep.py

```diff
@@ -161,3 +164,23 @@ def flush_storage(self):
         annotation = self.get_annotation()
         if annotation.get(SYNC_STORAGE) is not None:
             del annotation[SYNC_STORAGE]
+
+    def _fetch_missing_parents(self, item):
+        """
+        If data was fetched with portal type filter, this method will be used
+        to fill the missing parents for fetched objects.
+        :return:
+        """
+        parent_path = item.get("parent_path")
+        # Skip if the parent is portal object
+        if len(parent_path.split("/")) < 3:
+            return
```

Comment:
DRY;) Make a method that returns you the portal path and use it here as well for your check